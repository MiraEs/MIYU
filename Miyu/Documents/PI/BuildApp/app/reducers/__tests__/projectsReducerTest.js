'use strict';

jest.unmock('../../../app/reducers/projectsReducer');

import projectsReducer from '../projectsReducer';

import {
	CHANGE_PROJECT_STATUS_SUCCESS,
	CHANGE_PROJECT_STATUS_FAIL,
	DELETE_PROJECT_TEAM_MEMBER_SUCCESS,
	LOADING_PROJECTS,
	LOADING_SHOPPING_LISTS,
	RECEIVE_PROJECTS_FAIL,
	RECEIVE_PROJECTS_SUCCESS,
	SAVE_PROJECT_FAIL,
	SAVE_PROJECT_SUCCESS,
	SIGN_USER_OUT,
	UPDATE_PROJECT_SUCCESS,
} from '../../constants/constants' ;
import {
	UPDATE_PROJECTS_FILTER,
	PROJECTS_SEARCH_FILTER,
	SAVING_SHOPPING_LIST,
	SAVE_SHOPPING_LIST_SUCCESS,
	SAVE_SHOPPING_LIST_FAIL,
	CREATE_PROJECT_WITH_DEFAULT_GROUP_SUCCESS,
	ITEM_ADDED_TO_PROJECT,
	RESET_SHOPPING_LISTS_EDITING_FLAG,
	RESET_LOADING_FLAG,
} from '../../constants/ProjectConstants';

const initialState = {
	projects: {
		active: {
			myProjects: [],
			sharedProjects: [],
		},
		archived: {
			myProjects: [],
			sharedProjects: [],
		},
		preAuthProjects: [],
	},
	filteredProjects: {},
	isFiltering: false,
	error: '',
	isLoading: true,
	shoppingLists: [{
		project: {
			id: 0,
		},
		shoppingLists: [],
	}],
	filteredShoppingLists: [],
	itemAddedToProject: {
		added: false,
		projectId: null,
	},
};

describe('projectsReducer reducer', () => {

	it('should return initialState', () => {
		expect(projectsReducer(undefined, {})).toMatchSnapshot();
	});


	it('should CHANGE_PROJECT_STATUS_SUCCESS', () => {
		const action = {
			type: CHANGE_PROJECT_STATUS_SUCCESS,
			payload: [
				{
					myProjects: [],
					sharedProjects: [],
				},
				{
					myProjects: [],
					sharedProjects: [],
				},
			],
		};
		const state = projectsReducer(initialState, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should DELETE_PROJECT_TEAM_MEMBER_SUCCESS', () => {
		const action = {
			type: DELETE_PROJECT_TEAM_MEMBER_SUCCESS,
		};
		const state = projectsReducer(initialState, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should RECEIVE_PROJECTS_FAIL', () => {
		const action = {
			type: RECEIVE_PROJECTS_FAIL,
		};
		const state = projectsReducer(initialState, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should RECEIVE_PROJECTS_SUCCESS', () => {
		const action = {
			type: RECEIVE_PROJECTS_SUCCESS,
			payload: [
				{
					myProjects: [],
					sharedProjects: [],
				},
				{
					myProjects: [],
					sharedProjects: [],
				},
			],
		};
		const state = projectsReducer(initialState, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should SAVE_PROJECT_FAIL', () => {
		const action = {
			type: SAVE_PROJECT_FAIL,
		};
		const state = projectsReducer(initialState, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should SAVE_PROJECT_SUCCESS', () => {
		const action = {
			type: SAVE_PROJECT_SUCCESS,
			payload: [],
		};
		const state = projectsReducer(initialState, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should UPDATE_PROJECT_SUCCESS', () => {
		const action = {
			type: UPDATE_PROJECT_SUCCESS,
			payload: [],
		};
		const state = projectsReducer(initialState, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should UPDATE_PROJECTS_FILTER', () => {
		const action = {
			type: UPDATE_PROJECTS_FILTER,
			payload: [],
		};
		const state = projectsReducer(initialState, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should CHANGE_PROJECT_STATUS_FAIL', () => {
		const action = {
			type: CHANGE_PROJECT_STATUS_FAIL,
			payload: 'test',
		};
		const state = projectsReducer(initialState, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should SIGN_USER_OUT', () => {
		const action = {
			type: SIGN_USER_OUT,
		};
		const state = projectsReducer(initialState, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should LOADING_PROJECTS', () => {
		const action = {
			type: LOADING_PROJECTS,
			payload: true,
		};
		const state = projectsReducer(initialState, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should LOADING_SHOPPING_LISTS', () => {
		const action = {
			type: LOADING_SHOPPING_LISTS,
			payload: true,
		};
		const state = projectsReducer(initialState, action);
		expect(
			state
		).toMatchSnapshot();
	});

	it('should PROJECTS_SEARCH_FILTER', () => {
		const action = {
			type: PROJECTS_SEARCH_FILTER,
			term: '',
		};
		const state = projectsReducer(initialState, action);
		expect(state).toMatchSnapshot();
	});

	it('should SAVING_SHOPPING_LIST', () => {
		const action = {
			type: SAVING_SHOPPING_LIST,
		};
		const state = projectsReducer(initialState, action);
		expect(state).toMatchSnapshot();
	});

	it('should SAVE_SHOPPING_LIST_SUCCESS', () => {
		const action = {
			type: SAVE_SHOPPING_LIST_SUCCESS,
			payload: {
				projectId: 0,
			},
		};
		const state = projectsReducer(initialState, action);
		expect(state).toMatchSnapshot();
	});

	it('should SAVE_SHOPPING_LIST_FAIL', () => {
		const action = {
			type: SAVE_SHOPPING_LIST_FAIL,
			payload: {},
		};
		const state = projectsReducer(initialState, action);
		expect(state).toMatchSnapshot();
	});

	it('should CREATE_PROJECT_WITH_DEFAULT_GROUP_SUCCESS', () => {
		const action = {
			type: CREATE_PROJECT_WITH_DEFAULT_GROUP_SUCCESS,
			payload: {},
		};
		const state = projectsReducer(initialState, action);
		expect(state).toMatchSnapshot();
	});

	it('should ITEM_ADDED_TO_PROJECT', () => {
		const action = {
			type: ITEM_ADDED_TO_PROJECT,
			payload: {
				added: true,
				projectId: 0,
			},
		};
		const state = projectsReducer(initialState, action);
		expect(state).toMatchSnapshot();
	});

	it('should RESET_SHOPPING_LISTS_EDITING_FLAG', () => {
		const action = {
			type: RESET_SHOPPING_LISTS_EDITING_FLAG,
			payload: {
				projectId: 0,
			},
		};
		const state = projectsReducer(initialState, action);
		expect(state).toMatchSnapshot();
	});

	it('should RESET_LOADING_FLAG', () => {
		const action = {
			type: RESET_LOADING_FLAG,
			payload: {
				isLoading: true,
			},
		};
		const state = projectsReducer(initialState, action);
		expect(state).toMatchSnapshot();
	});
});
