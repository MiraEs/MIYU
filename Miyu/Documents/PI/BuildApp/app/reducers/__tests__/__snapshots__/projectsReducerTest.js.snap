// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`projectsReducer reducer should CHANGE_PROJECT_STATUS_FAIL 1`] = `
Object {
  "error": "test",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [
        Object {},
      ],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should CHANGE_PROJECT_STATUS_SUCCESS 1`] = `
Object {
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should CREATE_PROJECT_WITH_DEFAULT_GROUP_SUCCESS 1`] = `
Object {
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [
        Object {},
      ],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {},
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should DELETE_PROJECT_TEAM_MEMBER_SUCCESS 1`] = `
Object {
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should ITEM_ADDED_TO_PROJECT 1`] = `
Object {
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": true,
    "projectId": 0,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [
        Object {},
      ],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should LOADING_PROJECTS 1`] = `
Object {
  "aggregateData": Object {},
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "loadingShoppingLists": true,
  "projects": Object {
    "active": Object {
      "invitedProjects": Array [],
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "invitedProjects": Array [],
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [],
}
`;

exports[`projectsReducer reducer should LOADING_SHOPPING_LISTS 1`] = `
Object {
  "aggregateData": Object {},
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "loadingShoppingLists": true,
  "projects": Object {
    "active": Object {
      "invitedProjects": Array [],
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "invitedProjects": Array [],
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [],
}
`;

exports[`projectsReducer reducer should PROJECTS_SEARCH_FILTER 1`] = `
Object {
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [
        Object {},
      ],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should RECEIVE_PROJECTS_FAIL 1`] = `
Object {
  "error": "Unable to retrieve projects.",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": false,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should RECEIVE_PROJECTS_SUCCESS 1`] = `
Object {
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": false,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should RESET_LOADING_FLAG 1`] = `
Object {
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [
        Object {},
      ],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should RESET_SHOPPING_LISTS_EDITING_FLAG 1`] = `
Object {
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [
        Object {},
      ],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should SAVE_PROJECT_FAIL 1`] = `
Object {
  "error": undefined,
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should SAVE_PROJECT_SUCCESS 1`] = `
Object {
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [
        Array [],
      ],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should SAVE_SHOPPING_LIST_FAIL 1`] = `
Object {
  "error": "Failed to save the group",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "isSavingShoppingList": false,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [
        Object {},
      ],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should SAVE_SHOPPING_LIST_SUCCESS 1`] = `
Object {
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "isSavingShoppingList": false,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [
        Object {},
      ],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [
        Object {
          "projectId": 0,
        },
      ],
    },
  ],
}
`;

exports[`projectsReducer reducer should SAVING_SHOPPING_LIST 1`] = `
Object {
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "isSavingShoppingList": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [
        Object {},
      ],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should SIGN_USER_OUT 1`] = `
Object {
  "aggregateData": Object {},
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "loadingShoppingLists": true,
  "projects": Object {
    "active": Object {
      "invitedProjects": Array [],
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "invitedProjects": Array [],
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [],
}
`;

exports[`projectsReducer reducer should UPDATE_PROJECT_SUCCESS 1`] = `
Object {
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [
        Object {},
      ],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should UPDATE_PROJECTS_FILTER 1`] = `
Object {
  "error": "",
  "filteredProjects": Object {
    "active": Object {
      "myProjects": Array [
        Object {},
      ],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
  },
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "projects": Object {
    "active": Object {
      "myProjects": Array [
        Object {},
      ],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [
    Object {
      "project": Object {
        "id": 0,
      },
      "shoppingLists": Array [],
    },
  ],
}
`;

exports[`projectsReducer reducer should return initialState 1`] = `
Object {
  "aggregateData": Object {},
  "error": "",
  "filteredProjects": Object {},
  "filteredShoppingLists": Array [],
  "isFiltering": false,
  "isLoading": true,
  "itemAddedToProject": Object {
    "added": false,
    "projectId": null,
  },
  "loadingShoppingLists": true,
  "projects": Object {
    "active": Object {
      "invitedProjects": Array [],
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "archived": Object {
      "invitedProjects": Array [],
      "myProjects": Array [],
      "sharedProjects": Array [],
    },
    "preAuthProjects": Array [],
  },
  "shoppingLists": Array [],
}
`;
